Atomic/syscontainers.py

```diff
@@ -634,34 +634,54 @@ def _resolve_remote_path(self, remote_path):
+        """
+        Wrapper function that groups parameters into a dictionary for better readbility
+
+        returns the same result that _checkout returns
+        """
+        options = {"name" : name,
+                   "img" : img,
+                   "deployment" : deployment,
+                   "UpgradeMode" : mode,
+                   "values" : values,
+                   "destination" : destination,
+                   "extract_only" : extract_only,
+                   "remote" : remote,
+                   "prefix" : prefix,
+                   "InstalledFilesChecksum" : installed_files_checksum,
+                   "system_package" : system_package}
+        return self._checkout(repo, options)
+
+    def _checkout(self, repo, options):
+        options["destination"] = options["destination"] or os.path.join(self._get_system_checkout_path(), "{}.{}".format(options["name"], options["deployment"]))
+        unitfileout, tmpfilesout = self._get_systemd_destination_files(options["name"], options["prefix"])
+        options["unitfileout"] = unitfileout
+        options["tmpfilesout"] = tmpfilesout
 
-        if mode == SystemContainers.CHECKOUT_MODE_INSTALL:
+        install_mode = options["UpgradeMode"]  == SystemContainers.CHECKOUT_MODE_INSTALL
+
+        if install_mode:
             for f in [unitfileout, tmpfilesout]:
                 if os.path.exists(f):
                     raise ValueError("The file %s already exists." % f)
 
         try:
-            return self._do_checkout(repo, name, img, mode, deployment, values, destination, unitfileout, tmpfilesout, extract_only, remote, prefix,
-                                     installed_files_checksum=installed_files_checksum, system_package=system_package)
+            return self._do_checkout(repo, options)
         except (GLib.Error, ValueError, OSError, subprocess.CalledProcessError, KeyboardInterrupt) as e:
-            try:
-                if not extract_only and mode == SystemContainers.CHECKOUT_MODE_INSTALL:
-                    shutil.rmtree(destination)
-            except OSError:
-                pass
-            try:
-                if not extract_only and mode == SystemContainers.CHECKOUT_MODE_INSTALL:
+            if not options["extract_only"] and install_mode:
+                try:
+                    shutil.rmtree(options["destination"])
+                except OSError:
+                    pass
+                try:
                     os.unlink(unitfileout)
-            except OSError:
-                pass
-            try:
-                if not extract_only and mode == SystemContainers.CHECKOUT_MODE_INSTALL:
+                except OSError:
+                    pass
+                try:
                     os.unlink(tmpfilesout)
-            except OSError:
-                pass
+                except OSError:
```

Comment:
tabs here

Reply:
Thanks for reviewing =). 
> tabs here

hmm, can you explain a bit more on this? Are you referring to the "unnecessarily added tabs" here? Do note I have changed the expression here a bit, "I refactored the three duplicated if statements outside of the try except blocks", so each `try-except` block is now under one `if statement` umbrella.

Reply:
github showed there is a tab to indent the block.  If I misread it, please just ignore this comment :-)